src/cart/crt_rpc.c

```diff
@@ -1477,10 +1503,49 @@ crt_rpc_common_hdlr(struct crt_rpc_priv *rpc_priv)
 {
 	struct crt_context	*crt_ctx;
 	int			 rc = 0;
+	bool			skip_check = false;
+	d_rank_t		self_rank;
+
 
 	D_ASSERT(rpc_priv != NULL);
 	crt_ctx = rpc_priv->crp_pub.cr_ctx;
 
+	self_rank = crt_gdata.cg_grp->gg_srv_pri_grp->gp_self;
+
+	if (self_rank == CRT_NO_RANK)
+		skip_check = true;
+
+	/* Skip check when CORPC is sent to self */
+	if (rpc_priv->crp_coll) {
+		d_rank_t pri_root;
+
+		pri_root = grp_priv_get_primary_rank(
+				rpc_priv->crp_corpc_info->co_grp_priv,
+				rpc_priv->crp_corpc_info->co_root);
+
+		if (pri_root == self_rank)
+			skip_check = true;
+	}
+
+	if ((self_rank != rpc_priv->crp_req_hdr.cch_rank) ||
+		(crt_ctx->cc_idx != rpc_priv->crp_req_hdr.cch_tag)) {
+
+		if (!skip_check) {
+			D_ERROR("Mismatch rpc: %p opc: %x rank:%d tag:%d,"
```

Comment:
Is this really an error?   CaRT can get fairly chatty on errors.   If it can happen under normal circumstances and we handle it, it probably shouldn't be an error.